exercises/custom-set/.meta/gen.go

```diff
@@ -40,66 +45,76 @@ func strSlice(ns []int) []string {
 	return s
 }
 
-// The JSON structure we expect to be able to unmarshal into
-type js struct {
-	IsEmpty      unaryBool `json:"empty"`
-	Contains     eleBool
-	Subset       binaryBool
-	Disjoint     binaryBool
-	Equal        binaryBool
-	Add          eleOp
-	Intersection binaryOp
-	Difference   binaryOp
-	Union        binaryOp
+// istrSlice converts a slice of interface{} values whose
+// underlying members should be float64(JSON decoded integers)
+// to a slice of string.
+func istrSlice(ns []interface{}) []string {
+	s := make([]string, len(ns))
+	for i, n := range ns {
+		if fn, ok := n.(float64); ok {
+			s[i] = str(int(fn))
+		}
+	}
+	return s
 }
 
-type unaryBool struct {
-	Description string
-	Cases       []struct {
-		Description string
-		Set         []int
-		Expected    bool
+// dict sets up and returns a map for pairs of items given;
+// it is used in order to pass multiple parameters in a template call.
+func dict(values ...interface{}) (map[string]interface{}, error) {
+	if len(values)%2 != 0 {
+		return nil, errors.New("invalid dict call")
```

Comment:
is it worth having a more explicit error message? `invalid dict call: odd number of values given` or somesuch?